build_dcos_launch.sh

```diff
@@ -1,32 +1,19 @@
 #!/bin/bash
 #
 # Simple helper script to build dcos-launch binary
-# NOTE: this needs to be kept in sync with gen.build_deploy.bash::make_dcos_launch()
-
+# Note: must be run after this repo has been prepped with prep_local or prep_teamcity
 set -x
 set -o errexit -o pipefail
-
-# Cleanup from previous build
-rm -rf /tmp/dcos_build_venv
-
 # Force Python stdout/err to be unbuffered.
 export PYTHONUNBUFFERED="notemtpy"
-
-# Create a python virtual environment to install the DC/OS tools to
-python3 -m venv /tmp/dcos_launch_venv
-. /tmp/dcos_launch_venv/bin/activate
-
-# Make a clean as possible clone
-if [[ -n $(git -C $PWD status --porcelain -uno -z) ]]; then
-  echo "Commit all changes before attempting to build!";
-  exit 1;
-fi
-rm -rf /tmp/dcos-installer-build
-git clone -q "file://$PWD" /tmp/dcos-installer-build/
-pushd /tmp/dcos-installer-build
-# Install the DC/OS tools
-pip install -e /tmp/dcos-installer-build
-cp launch/dcos-launch.spec ./
-pyinstaller --log-level=DEBUG dcos-launch.spec
+pushd packages/dcos-launch
+new_package_name=`mkpanda | tail -n 1 | cut -d':' -f2`
 popd
-cp /tmp/dcos-installer-build/dist/dcos-launch ./
+mkdir -p tmp/dcos_build
+cp $new_package_name tmp/dcos_build/dcos-launch.tar.xz
+pushd tmp/dcos_build
+xz -d dcos-launch.tar.xz
+tar xf dcos-launch.tar
+popd
+cp tmp/dcos_build/dcos-launch .
+rm -rf tmp/dcos_build/
```

Comment:
My mistake, recent versions of `tar` auto-detect compression format these days. You can use `-J` or `--xz` to explicitly decompress with `xz` (added to `tar` in 2009, doubt we'll run into a version that doesn't support it). https://www.gnu.org/software/tar/manual/tar.html#SEC136